### PR TITLE
add parsing from sql to dynamo where clause

### PR DESCRIPTION
**Pull request by nofarpepperi**

### CODE REVIEW

File: filters/string-filter.ts [hunk 1]
```diff
@@ -100,4 +101,102 @@ export class StringFilter extends Filter {
...
+
+        switch (this.operation) {
+            case 'IsEmpty':
+                var markValue = ":" + letterForMark + count;
+                res.ExpressionAttributeValues[markValue] = ''
+                var markName = "#" + letterForMark + count;
+                res.ExpressionAttributeNames[markName] = this.apiName
+                count++;
+                res.ResString = `attribute_not_exists (${markName}) OR ${markName} = ${markValue}`;
+                res.Count = count;
```
Comment (Chasky-H) [inline filters/string-filter.ts:116]: It seems like this can be refactoring into a helper function, or at least part of it.

**Review by Chasky-H — CHANGES_REQUESTED** [review]

File: filters/filter.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { Query } from 'elastic-builder';
+import { DynamoResultObject } from './DynamoObjectResult';
 
 export default abstract class Filter {
     constructor(protected apiName: string) {}
 
     abstract apply(value: any): boolean;
     abstract toSQLWhereClause(): string;
     abstract toKibanaFilter(): Query;
+    abstract toDynamoWhereClause(letterForMark: string, expressionAttributeNames: any, expressionAttributeValues: any, count: number) : DynamoResultObject;
```
[the remark this answers is not in the document]
Comment (nofarpepperi) [reply filters/filter.ts]: changed to toDynamoDBQuery

File: filters/filter-collection.ts [hunk 3]
```diff
@@ -64,4 +65,30 @@ export default class FilterCollection extends Filter {
 
         return boolQuery;
     }
+
+    toDynamoWhereClause(letterForMark: string, expressionAttributeNames: any, expressionAttributeValues: any, count: number): DynamoResultObject {
+        let res = new DynamoResultObject(count, expressionAttributeNames, expressionAttributeValues, '');
+
+        this.filters.forEach((filter) => {
+            count = res.Count;
+            var DynamoWhereClause = filter.toDynamoWhereClause(letterForMark, expressionAttributeNames, expressionAttributeValues, count);
```
[the remark this answers is not in the document]
Comment (nofarpepperi) [reply filters/filter-collection.ts]: fixed

File: filters/filter-factory.ts [hunk 4]
```diff
@@ -63,4 +63,53 @@ export class FilterFactory {
...
+                        apiName,
+                        operation as NumberOperation,
+                        values.map((x) => +x),
+                    );
+                }
+
+                case 'Double': {
+                    return new NumberFilter(apiName, operation as NumberOperation, values.map(parseFloat));
+                }
+
```
[the remark this answers is not in the document]
Comment (nofarpepperi) [reply filters/filter-factory.ts]: fixed

File: index.ts [hunk 5]
```diff
@@ -94,6 +95,15 @@ export function toKibanaQueryJSON(jsonFilter: JSONFilter | undefined) {
     throw new Error('jsonFilter is a mandatory parameter');
 }
 
+export function toDynamoQuery(jsonFilter: JSONFilter | undefined, letterForMark: string, expressionAttributeNames: any, expressionAttributeValues: any, count: number): DynamoResultObject {
```
[the remark this answers is not in the document]
Comment (nofarpepperi) [reply index.ts]: fixed

File: filters/string-filter.ts (hunk 1 above)
Comment (nofarpepperi) [reply filters/string-filter.ts]: fixed

**Review by Chasky-H — APPROVED** [review]

**Pull request merged**